src/main/java/org/springframework/data/couchbase/core/mapping/id/GenerationStrategy.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017 the original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.data.couchbase.core.mapping.id;
+
+/**
+ * Keys generation strategy
+ *
+ * @author Subhashni Balakrishnan
+ */
+public enum GenerationStrategy {
+	/**
+	 * Constructs key from the entity attributes using the supplied id, prefix, suffix
+	 */
+	USE_ATTRIBUTES,
+
+	/**
+	 * Uses an uuid generator
+	 */
+	UNIQUE
```

Comment:
have you guys thought about another "unique" strategy based on a counter document?